mrmustard/lab_dev/wires.py

```diff
@@ -108,110 +119,170 @@ def __init__(
 
     @classmethod
-    def _from_data(cls, id_array: np.ndarray, modes: list[int], mask=None):
-        r"""Private class method to initialize Wires object from the given data."""
+    def _from_data(cls, id_array, modes, mask=None):
+        r"""
+        Initializes ``Wires`` object from its private attributes.
+        """
         w = cls()
         w._id_array = id_array
         w._modes = modes
         w._mask = mask if mask is not None else np.ones_like(w._id_array)
         return w
 
     def _view(self, masked_rows: tuple[int, ...] = (), masked_cols: tuple[int, ...] = ()) -> Wires:
-        r"""A masked view of this Wires object."""
+        r"""
+        A masked view of this Wires object.
+        """
         w = self._from_data(self._id_array, self._modes, self._mask.copy())
         w._mask[masked_rows, :] = -1
         w._mask[:, masked_cols] = -1
         return w
 
     def _mode(self, mode: int) -> np.ndarray:
-        "A slice of the id_array matrix at the given mode."
+        r"""
+        A slice of the id_array matrix at the given mode.
+        """
         return np.maximum(0, self.id_array[[self._modes.index(mode)]])[0]
 
     @property
     def id_array(self) -> np.ndarray:
-        "The id_array of the available wires in the standard order (bra/ket x out/in x mode)."
+        r"""
+        The id_array of the available wires in a two-dimensional array, where line ``j`` contains
+        the ids (in the standard order) for mode ``j``.
+        """
         return self._id_array * self._mask
 
     @property
     def ids(self) -> list[int]:
-        "The list of ids of the available wires in the standard order."
+        r"""
+        The list of ids of the available wires in the standard order.
+        """
         flat = self.id_array.T.ravel()
         return flat[flat > 0].tolist()
 
     @ids.setter
     def ids(self, ids: list[int]):
-        "Sets the ids of the available wires."
+        r"""
+        Sets the ids of the available wires.
+
+        Args:
+            ids: The new ids.
+
+        Raises:
+            ValueError: If the number of ids does not match the expected number.
+        """
         if len(ids) != len(self.ids):
             raise ValueError(f"wrong number of ids (expected {len(self.ids)}, got {len(ids)})")
         self._id_array.flat[self.id_array.flatten() > 0] = ids
 
     @property
     def modes(self) -> list[int]:
-        "The set of modes spanned by the populated wires."
+        r"""
+        The set of modes of the populated wires.
+        """
         return [m for m in self._modes if any(self.id_array[self._modes.index(m)] > 0)]
 
     @property
     def indices(self) -> list[int]:
-        r"""Returns the array of indices of this subset in the standard order.
-        (bra/ket x out/in x mode). Use this to get the indices for bargmann contractions.
+        r"""
+        The array of indices of this subset in the standard order.
```

Comment:
```suggestion
        The array of indices of this subset in the standard order, skipping over wires not belonging to the subset.
        .. code-block::
            >>> w = Wires(modes_in_ket = [0,1], modes_out_ket = [0,1])
            >>> assert w.indices == [0,1,2,3]
            >>> assert w.input.indices == [2,3]
```

Reply:
How about `The array of indices of this ``Wires`` in the standard order. When a subset is selected, it skips  the indices of wires that do not belong to the subset.`
Just because this applies to wires, not only to subsets